src/scripts.d/66_device_Network

```diff
@@ -0,0 +1,99 @@
+#!/bin/sh
+set -e
+
+EXIT_OK=0
+
+# All the network interfaces
+INTERFACES=$(ls /sys/class/net)
+
+get_lan_info() {
+    for IFACE in $INTERFACES; do
+        # Exclude loopback
+        if [ $IFACE = "lo" ]; then
+            continue
+        fi
+
+        MAC=$(cat /sys/class/net/"$IFACE"/address)
+        ENABLED=$(cat /sys/class/net/"$IFACE"/operstate | grep -q 'up' && echo 1 || echo 0)
+
+        IP4_LIST=$(ip -4 addr show dev "$IFACE" 2>/dev/null | awk '
+            $1 == "inet" {
+                ip_list = (ip_list ? ip_list "," : "") "\"" $2 "\"";
+            }
+            END {
+                print ip_list
+            }
+        ')
+
+        IP6_LIST=$(ip -6 addr show dev "$IFACE" 2>/dev/null | awk '
+            $1 == "inet6" && $2 !~ /^fe80/ {
+                ip_list = (ip_list ? ip_list "," : "") "\"" $2 "\"";
+            }
+            END {
+                print ip_list
+            }
+        ')
+
+        printf 'LAN_%s={"name":"%s","ip4":[%s],"ip6":[%s],"mac":"%s","enabled":%s}\n' "$IFACE" "$IFACE" "${IP4_LIST:-}" "${IP6_LIST:-}" "${MAC:-}" "${ENABLED:-0}"
+
+    done
+}
+
+get_wan_info() {
+    TARGET="8.8.8.8"  # PING target
+    COUNT=1           # PING count
+    DEADLINE=5        # PING deadline
+
+    if ! command -v ping >/dev/null 2>&1; then
+        PING_STATUS="Ping not found"
+    else
+        ping -c "$COUNT" -w "$DEADLINE" "$TARGET" > /dev/null 2>&1
+        EXIT_CODE=$?
+        if [ $EXIT_CODE -eq 0 ]; then
+            PING_STATUS="OK"
+        else
+            PING_STATUS="Failed"
+        fi
+    fi
+
+    if command -v curl >/dev/null 2>&1; then
+        GLOBAL_IP=$(curl -s --connect-timeout 5 ifconfig.me)
+    fi
+    printf 'WAN={"globalIp":"%s","pingStatus":"%s"}\n' "${GLOBAL_IP:-}" "${PING_STATUS}"
+}
+
+# Get system resolver DNS info
+get_system_dns_info() {
+    DNS_SERVERS=$(awk '
+            /^nameserver/ {
+                ip = "\"" $2 "\""
+                output = (output ? output "," : "") ip
+            }
+            END {
+                print output
+            }
+        ' /etc/resolv.conf )
+    printf 'systemResolver={"dns":[%s]}\n' "${DNS_SERVERS:-}"
+}
+
+# Get DHCP and upstream DNS info
+get_dhcp_info() {
+    # Try to find DHCP server config (example: dnsmasq)
+    if [ -f /etc/dnsmasq.conf ]; then
+        DNS1=$(grep -m1 '^server=' /etc/dnsmasq.conf | cut -d= -f2)
+        DNS2=$(grep -m2 '^server=' /etc/dnsmasq.conf | tail -n1 | cut -d= -f2)
+        DOMAIN=$(grep '^domain=' /etc/dnsmasq.conf | cut -d= -f2)
+        RANGE_START=$(grep '^dhcp-range=' /etc/dnsmasq.conf | cut -d= -f2 | cut -d, -f1)
+        RANGE_END=$(grep '^dhcp-range=' /etc/dnsmasq.conf | cut -d= -f2 | cut -d, -f2)
+
+        printf 'DHCP={"dns1":"%s","dns2":"%s","domainName":"%s","addressRange":{"start":"%s","end":"%s"}}\n' "${DNS1:-}" "${DNS2:-}" "${DOMAIN:--}" "${RANGE_START:-}" "${RANGE_END:-}"
```

Comment:
It would be helpful to add a check on the fields and only publish the twin info if there is at least one field set, as I ran it on a device and it output the following empty values (which is expected on this device)

```sh
DHCP={"dns1":"","dns2":"","domainName":"-","addressRange":{"start":"","end":""}}
```